content/developer/addons/addon-info.md

```diff
@@ -0,0 +1,278 @@
+---
+title: Addon Information
+tags:
+- Developers
+- Addons
+- Configuration
+- about.php
+- addon.json
+category: addons
+menu:
+  developer:
+    parent: addons
+aliases:
+- /developer/theming/about-php
+---
+## Addon Information
+
+Every Vanilla addons must declare certain information about itself. This includes:
+
+- Addon Metadata
+- Author Metadata
+- Addon support links
+- Addon configuration details
+- Details about how to load the addon
+
+### Which format do I use?
+
+Addon information can come in different formats be located in different places depending on which version of Vanilla you are running
+
+**I am using Vanilla Forums >= 2.4 or Vanilla Cloud**
+
+Use the new [addon.json](#the-addon-json-file) format.
+
+**I am using Vanilla Forums < 2.4**
+
+See [PHP Information Format](#php-information-format).
+
+## The `addon.json` file
+
+In an effort to simplify the management and creation of plugins, applications, and themes, Vanilla `2.4` and onwards uses a single unified addon information format. The new format is a `json` file that should live in the root of your addon's directory. It looks like this:
+
+### `addon.json` example
+
+```json
+{
+    "key": "example-addon",
+    "name": "Example Addon",
+    "description": "Provides an example of an Addon",
+    "version": "2.0.0",
+    "documentationUrl": "http://site.com/link-to-documentation",
+    "type": "addon",
+    "icon": "icon-location.png",
+    "mobileFriendly": true,
+    "hasLocale": false,
+    "settingsUrl": "/settings/myaddon",
+    "settingsPermission": "Garden.Settings.Manage",
+    "registerPermissions": {
+        "Plugins.MyPlugin.MyPermission": "Garden.Settings.Manage"
+    },
+    "authors": [
+        {
+            "name": "Todd Burry",
+            "email": "todd@vanillaforums.com",
+            "homepage": "https://open.vanillaforums.com/profile/todd"
+        }
+    ],
+    "require": {
+        "vanilla": ">=2.4"
+    }
+}
+```
+
+### Theme `addon.json` example
+
+Theme's tend to be a little bit simpler and would look like this:
+
+```json
+{
+    "key": "example-theme",
+    "name": "Example Theme",
+    "description": "Custom Theme Example",
+    "version": "2.0.0",
+    "type": "theme",
+    "license": "MIT",
+    "author": [
+        {
+            "name": "Adam Charron",
+            "email": "adam.c@vanillaforums.com"
+        }
+    ],
+    "layout": {
+        "categories": "Modern",
+        "discussions": "Modern"
+    }
+}
+```
+
+### Details
+
+#### key
+
+The theme key. This should be unique and ***must exactly match*** the folder name, including capitalization. So an addon with the key `example-addon` would need to be located in a `plugins/example-addon` directory in your Vanilla installation. Likewise a theme with the key `UpperCaseTheme` would need to be located in a `themes/UpperCaseTheme` directory inside your Vanilla installation.
```

Comment:
Since it will be used internally we should specify that we want lower case folders. (that's the new standard)